llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -984,32 +984,37 @@ void State::addInfoFor(BasicBlock &BB) {
       continue;
     }
 
-    if (match(&I, m_Intrinsic<Intrinsic::ssub_with_overflow>())) {
+    auto *II = dyn_cast<IntrinsicInst>(&I);
+    Intrinsic::ID ID = II ? II->getIntrinsicID() : Intrinsic::not_intrinsic;
+    switch (ID) {
+    case Intrinsic::assume:
```

Comment:
probably best to wrap the below below in `{}`

Reply:
Done